language/move-prover/bytecode-to-boogie/tests/driver/mod.rs

```diff
@@ -61,8 +59,54 @@ pub fn test(sources: &[&str]) {
     fs::write(&boogie_file_path, boogie_str).expect("cannot write boogie file");
 
     if env::var("BOOGIE_EXE").is_ok() {
-        // Call boogie to verify results. Right now, we call boogie for type checking only.
-        assert!(driver.call_boogie(&boogie_file_path));
+        // Call boogie and verify results.
+        let (_, diag) = driver
+            .call_boogie(&boogie_file_path)
+            .expect("boogie execution ok");
+        verify_diag(sources, diag);
+    }
+}
+
+/// Extracts expected diags from sources and compares it with actual diags.
+fn verify_diag(sources: &[&str], mut actual_diag: Vec<String>) {
```

Comment:
CC @vgao1996; this looks like an interesting and very lightweight way to implement some of the matching that the functional tests do.